sql/core/src/main/scala/org/apache/spark/sql/execution/SQLExecution.scala

```diff
@@ -39,6 +39,32 @@ object SQLExecution {
     executionIdToQueryExecution.get(executionId)
   }
 
+  private val testing = sys.props.contains("spark.testing")
+
+  private[sql] def checkSQLExecutionId(sparkSession: SparkSession): Unit = {
```

Comment:
To keep this PR from growing too big, I want to just use it where I've removed `withNewExecutionId` to check for regressions. I'll follow up with another PR with more checks.